decision-records/2021-02-07-selective-testing.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Selective testing
+who: @fortmarek
+---
+
+## Context
+
+> What motivated us to implement this feature
+
+Re-running all the tests on the CI can take quite a long time on larger projects - we should be able to recognize what has changed and run tests only for those changed modules.
+
+## What
+
+> What's selective testing? Who would benefit from this feature? Why would you use selective testing in your workflows?
+
+Selective testing is bundled in `tuist test` command - it re-runs only tests for the modules (or their transitive dependencies) that have changed. If your tests are taking a long time, you might want to use `test` command to strip the tests time tremendously - although, this will only work effectively if your project is modular as that is the granularity we use.
+
+## How
+
+> Overview of how it's been implemented.
+
+`tuist test` first hashes the graph targets that can be tested (directly or transitively). It then "tree-shakes" the graph removing all the targets from test actions that have not changed from the last successful test run. Afterwards, tests are run. If they are successful, we save the hashes computed early in the process to `$TUIST_CACHE_DIRECTORY/TestsCache` which is by default at `~/.tuist/Cache/TestsCache`. We have been able to reuse a lot of functionality from `tuist cache`.
```

Comment:
I'd not mention `$TUIST_CACHE_DIRECTORY` because it's an internal variable that allows us to control the directory for testing purposes.

Reply:
`TUIST_CACHE_DIRECTORY` is actually now [documented](https://github.com/tuist/tuist/pull/2450#issuecomment-774531386) to allow users for custom location of the cache on CI